Tasks/SonarQubePostTest/SonarQubeReportHandler.ps1

```diff
@@ -25,37 +27,113 @@ function UploadLegacySummaryMdReport
 	$sonarQubeOutDir = GetSonarQubeOutDirectory
 
 	# Upload the summary markdown file
-	$summaryMdPath = [System.IO.Path]::Combine($sonarQubeOutDir, "summary.md")
-	Write-Verbose "Looking for a summary report at $summaryMdPath"
+	$legacySummaryReportPath = [System.IO.Path]::Combine($sonarQubeOutDir, "summary.md")
+	Write-Verbose "Looking for a summary report at $legacySummaryReportPath"
 
-	if ([System.IO.File]::Exists($summaryMdPath))
+	if ([System.IO.File]::Exists($legacySummaryReportPath))
 	{
-		Write-Verbose "Uploading the summary.md file"
-        Write-Host "##vso[task.addattachment type=Distributedtask.Core.Summary;name=SonarQube Analysis Report;]$summaryMdPath"
+         UploadReport $legacySummaryReportPath
 	}
 	else
 	{
-		 Write-Warning "Could not find the summary report file $summaryMdPath"
+		 Write-Warning "Could not find the summary report file $legacySummaryReportPath"
 	}
 }
 
 function UploadSummaryReportInternal
 {
     Assert ((IsReportEnabled) -eq $true) "The summary report is disabled"
     Assert ((IsPrBuild) -eq $false) "Cannot produce a report because the analysis was done in issues mode"
-        
-    $qualityGateStatus = GetOrFetchQualityGateStatus
-        
-    #TODO: actually upload a report
-    Write-Host "The quality gate status is $qualityGateStatus"
+
+    Write-Host "Creating a summary report"        
+    $reportPath = CreateReport
+    Write-Host "Uploading the report"
+    UploadReport $reportPath
 }
 
 function IsReportEnabled
 {
-    $includeReport = GetTaskContextVariable "MSBuild.SonarQube.IncludeReport"
+    $includeReport = GetTaskContextVariable "MSBuild.SonarQube.Internal.IncludeFullReport"
     $reportEnabled = [System.Convert]::ToBoolean($includeReport)
     
     return $reportEnabled
 }
 
+function UploadReport
+{
+    param($reportPath)
+    
+    Write-Host "##vso[task.addattachment type=Distributedtask.Core.Summary;name=SonarQube Analysis Report;]$reportPath"
+}
+
+function CreateReport
+{
+    $dashboardUri = GetTaskContextVariable "MSBuild.SonarQube.ProjectUri"
+    
+    Assert (![String]::IsNullOrEmpty($dashboardUri)) "Could not find the dashboard uri"
+    
+    $qualityGateStatus = GetOrFetchQualityGateStatus
+    $qualityGateColor = "" 
+    $qualityGateLabel = ""
+    GetQualityGateVisualDetails $qualityGateStatus ([ref]$qualityGateColor) ([ref]$qualityGateLabel)
+    
+    $reportTemplate  = '<div style="padding:5px 0px">
+<span>Quality Gate</span>
+<span style="padding:4px 10px; background-color:{0}; color:#fff; display:inline-block">{1}</span>
+</div>
+<div>
+<a target="_blank" href="{2}">Detailed report ></a>
+</div>'
+    
+    $reportContents = [String]::Format($reportTemplate, $qualityGateColor, $qualityGateLabel, $dashboardUri)    
+    $sonarQubeOutDir = GetSonarQubeOutDirectory
+    $reportPath = [IO.Path]::Combine($sonarQubeOutDir, "newSummaryReport.md");
+    
+    [IO.File]::WriteAllText($reportPath, $reportContents)
```

Comment:
I am placing the new report in the same directory as the old one. WriteAllText will overwrite any existing files, although that should not happen because the .sonarqube directory is cleared out each analysis.